src/Sylius/Bundle/CoreBundle/Applicator/ProductCatalogPromotionApplicatorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Applicator;
+
+use Sylius\Component\Core\Model\ProductInterface;
+
+interface ProductCatalogPromotionApplicatorInterface
+{
+    public function applyPercentageDiscount(ProductInterface $product, float $discount): void;
```

Comment:
From an interface point of view, this product catalog promotion applicator will be only related with % discounts. What kind of different implementation were you thinking of?